plugin/path_config.go

```diff
@@ -77,6 +92,45 @@ func (b *backend) pathConfigWrite(ctx context.Context, req *logical.Request, dat
 		cfg.CredentialsRaw = credentialsRaw.(string)
 	}
 
+	// set plugin identity token fields
+	if err := cfg.ParsePluginIdentityTokenFields(data); err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	// set Service Account email
+	saEmail, ok := data.GetOk("service_account_email")
+	if ok {
+		cfg.ServiceAccountEmail = saEmail.(string)
+	}
+
+	if cfg.IdentityTokenAudience != "" && cfg.CredentialsRaw != "" {
+		return logical.ErrorResponse("only one of 'credentials' or 'identity_token_audience' can be set"), nil
+	}
+
+	if cfg.IdentityTokenAudience != "" && cfg.ServiceAccountEmail == "" {
+		return logical.ErrorResponse("missing required 'service_account_email' when 'identity_token_audience' is set"), nil
+	}
+
+	// generate token to check if WIF is enabled on this edition of Vault
+	if cfg.IdentityTokenAudience != "" {
+		_, err := b.System().GenerateIdentityToken(ctx, &pluginutil.IdentityTokenRequest{
+			Audience: cfg.IdentityTokenAudience,
+		})
+		if err != nil {
+			if errors.Is(err, pluginidentityutil.ErrPluginWorkloadIdentityUnsupported) {
+				return logical.ErrorResponse(err.Error()), nil
+			}
+			return nil, err
+		}
```

Comment:
I wonder if we should wrap all this in a `pluginidentityutil` helper? It looks like this will be identical for all the plugins that support wif.

Reply:
I just looked into it briefly and it looks like what we have here is the best we can do since we want to return a `logical.ErrorResponse` in the case of ErrPluginWorkloadIdentityUnsupported.